### PR TITLE
i18n(zh-cn): Translate tutorial

### PR DESCRIPTION
**Pull request by KurumiFrost**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- Tutorial introduction Chinese translate

### CODE REVIEW

**Comment by liruifengv** [comment]
I found a bug, maybe a feature.

image

The Tutorial Tracker and the unit index on the right are incorrect until all the units have been translated.

File: src/content/docs/zh-cn/tutorial/0-introduction/1.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+(这些数据只保存在你的浏览器的本地存储中，在其他地方无法使用。没有数据会被发送到 Astro，也不会被 Astro 存储）。
+</details>
+
+<details>
+<summary>我已经知道第一单元该做什么了。我可以跳过吗？</summary>
+
+你可以使用[Unit 1](/en/tutorial/1-setup/)里面的课程，确保你拥有完成教程所需的开发工具和账户。它将指导你创建一个新的 Astro 项目，将其存储在 GitHub 上，并将其部署到 Netlify。
+
+如果你[创建一个新的、空的Astro项目](/en/install/auto/)并对你的设置感到满意，你可以安全地跳到[第二单元](/en/tutorial/2-pages/)，你将开始在你的项目中制作新页面。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/0-introduction/1.mdx:34]: Thanks hard work! @KurumiFrost .
There are some minor issues that need to be changed here.
```suggestion
如果你[已经创建了一个新的、空的Astro项目](/zh-cn/install/auto/)并对你的设置感到满意，你可以安全地跳到[第二单元](/en/tutorial/2-pages/)，你将开始在你的项目中制作新页面。
```

File: src/content/docs/zh-cn/tutorial/0-introduction/1.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+使用跟踪器是可选的，但如果你在多次访问中完成教程，它可以帮助你记住你的位置。你也可以把一些核对表留空，以提醒你以后要重温的单元。
+
+(这些数据只保存在你的浏览器的本地存储中，在其他地方无法使用。没有数据会被发送到 Astro，也不会被 Astro 存储）。
+</details>
+
+<details>
+<summary>我已经知道第一单元该做什么了。我可以跳过吗？</summary>
+
+你可以使用[Unit 1](/en/tutorial/1-setup/)里面的课程，确保你拥有完成教程所需的开发工具和账户。它将指导你创建一个新的 Astro 项目，将其存储在 GitHub 上，并将其部署到 Netlify。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/0-introduction/1.mdx:32]: `Unit 1` or `第一单元`, needs to be consistent below
```suggestion
你可以使用[第一单元](/zh-cn/tutorial/1-setup/)里面的课程，确保你拥有完成教程所需的开发工具和账户。它将指导你创建一个新的 Astro 项目，将其存储在 GitHub 上，并将其部署到 Netlify。
```

File: src/content/docs/zh-cn/tutorial/0-introduction/1.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+如果你对 **HTML**、**Markdown**、**CSS** 和一点 **JavaScript** 有一些基本的了解，那么你就可以开始了！ 只需按照说明进行操作，你就可以完成整个教程。 Astro 适合所有人！ 🧑‍🚀 👩‍🚀 👨‍🚀
+
+你还需要一个 [GitHub](https://github.com)（或类似的）账号来将你的项目部署到网络上。
+
+<details>
+<summary>如何使用每页底部的清单？</summary>
+
+你把它们勾掉!
+
+在每个页面的末尾，你会发现一个可点击的任务清单，你现在应该能够做到。勾选这些项目，就可以在 "教程追踪 "中看到你的进展。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/0-introduction/1.mdx:22]:
```suggestion
在每个页面的末尾，你会发现一个可点击的任务清单，其中列出了你现在应该能够完成的任务。勾选这些项目，就可以在 "教程追踪 "中看到你的进展。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

File: src/content/docs/zh-cn/tutorial/0-introduction/1.mdx (hunk 2 above)
Comment (KurumiFrost) [reply src/content/docs/zh-cn/tutorial/0-introduction/1.mdx]: Thanks for your careful review @liruifengv. I resolved this.

File: src/content/docs/zh-cn/tutorial/0-introduction/1.mdx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+(这些数据只保存在你的浏览器的本地存储中，在其他地方无法使用。没有数据会被发送到 Astro，也不会被 Astro 存储）。
+</details>
+
+<details>
+<summary>我已经知道第一单元该做什么了。我可以跳过吗？</summary>
+
+你可以使用[第一单元](/zh-cn/tutorial/1-setup/)里面的课程，确保你拥有完成教程所需的开发工具和账户。它将指导你创建一个新的 Astro 项目，将其存储在 GitHub 上，并将其部署到 Netlify。
+
+如果你[已经创建了一个新的、空的Astro项目](/zh-cn/install/auto/)并对你的设置感到满意，你可以安全地跳到[第二单元](/en/tutorial/2-pages/)，你将开始在你的项目中制作新页面。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/tutorial/0-introduction/1.mdx:34]: There can be `zh-cn`. You can use github's commit suggestion function directly.
```suggestion
如果你[已经创建了一个新的、空的Astro项目](/zh-cn/install/auto/)并对你的设置感到满意，你可以安全地跳到[第二单元](/zh-cn/tutorial/2-pages/)，你将开始在你的项目中制作新页面。
```

Comment (KurumiFrost) [reply src/content/docs/zh-cn/tutorial/0-introduction/1.mdx]: > There can be `zh-cn`. You can use github's commit suggestion function directly.

Thanks a lot, I didn't notice this before.
Now it's outdated which can not be applied.
I'll use this feature on my after contribute.

**Comment by liruifengv** [comment]
Build failed. Congratulations, you encountered two bugs in your first contribution. @KurumiFrost

**Comment by KurumiFrost** [comment]
> Build failed. Congratulations, you encountered two bugs in your first contribution. @KurumiFrost

Lol, why I'm so lucky XD!

**Comment by Yan-Thomas** [comment]
Haha, nice bug finding here, thanks @KurumiFrost & @liruifengv. The tutorial isn't meant to be translated yet though, that's why the errors, I will take a look at the issues and PR the necessary changes for it to work properly. For now, I'll mark this PR as a draft, and I'm sorry for doing this in your first one @KurumiFrost, but hey, there's still time for you to PR an i18n translation and be our 400th contributor 👀!

**Comment by sarah11918** [comment]
Hi @KurumiFrost and @liruifengv !

Wow, I see how much work you put into this already! It's so exciting that you want to translate the tutorial!

We will have to discuss with the Docs team how we want to handle tutorial translations. As @Yan-Thomas said, the tutorial is marked `false` for being ready for i18n contributions right now. In regular docs it's OK to have only some pages translated. But, for the tutorial it really doesn't make sense to have a translation published until the ENTIRE tutorial is translated. We cannot accept just some pages  because it could be confusing and misleading to people who will expect it to be available in their language.

I will leave this open as a draft for now, but we may decide to close this PR and ask you to instead keep working locally on your branch (if you want to!) and wait to open a PR until 1) We have prepared all the site navigation/infrastructure to support multiple languages for the tutorial and 2) the entire tutorial is translated. This might be a long time from now!

We will have to think about the best way to allow you to keep translating, and to have others be able to contribute to your work with feedback and suggestions, and whether we can support that in this docs repo right now, or whether we will suggest you work locally for a while first, and let others know how they can work with you.

Thank you for your understanding!

**Comment by KurumiFrost** [comment]
Hi~ @sarah11918 @Yan-Thomas @liruifengv

Thanks for your reply!
I fully understand! An incomplete tutorial definitely causes some misleading!!
Actually, this is my first time to contribute to docs translation. Even now, I already learned some GitHub tricks and leveled up my translation skill. So I'll be willing to put my time on further tutorial translation (maybe you can keep this PR till it fully translated). It may take some while 🌵.

**Comment by Yan-Thomas** [comment]
Thanks for the patience @KurumiFrost, we've fully updated our infrastructure to handle tutorial translations better, so you should be good to continue your translation at your own pace with the site building and preview deploying successfully 🙌

File: src/content/docs/zh-cn/tutorial/1-setup/3.mdx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        <meta name="generator" content = {Astro.generator} >
+        <title>Astro</title>
+      </head>
+      <body>
+        <h1>Astro</h1>
+      </body>
+    </html>
+    ```
+
+2. 编辑页面`<body>` 的内容。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/3.mdx:45]:
```suggestion
2. 编辑页面 `<body>` 的内容。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+:::note
+如果你已经熟悉 git 并拥有自己的工作流程，请使用你喜欢的方法为你的项目创建一个新的 GitHub 仓库。 跳到下一页：[“将站点部署到网上”](/cn/tutorial/1-setup/5/)。
+:::
+
+## 在 GitHub 上创建一个仓库
+
+尽管有几种方法可以将你的本地代码存储在 GitHub 中，但本教程将指导你完成一种不需要在命令行中使用 git 的方法。
+
+1. 在浏览器中登录 GitHub.com，点击屏幕右上角的<kbd>+</kbd>新建一个代码仓库`repository`。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:29]:
```suggestion
1. 在浏览器中登录 GitHub.com，点击屏幕右上角的 <kbd>+</kbd> 新建一个代码仓库（Repository）。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+2. 为你的仓库起一个名称。 不必与你的项目文件夹同名。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:31]:
```suggestion
2. 为你的仓库起一个名称。不必与你的项目文件夹同名。
```

Unnecessary extra spaces.

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+3. 你将看到一些选项，但无需更改任何默认设置。 向下滚动并单击按钮以<kbd>Create Repository</kbd>。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:33]:
```suggestion
3. 你将看到一些选项，但无需更改任何默认设置。向下滚动并单击按钮以 <kbd>Create Repository</kbd>。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+4. 你将看到各种后续的选项步骤，但你不需要更改其中的任何一个。 记下仓库的地址`URL`。 现在可以不执行任何操作然后退出页面。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:35]:
```suggestion
4. 你将看到各种后续的选项步骤，但你不需要更改其中的任何一个。记下仓库的地址（URL）。现在可以不执行任何操作然后退出页面。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+
+## 将本地代码提交到 GitHub
+
+在上一节中，你对页面内容进行了更改。 这意味着你的项目文件已更改，VS Code 应在“源”菜单`Source Control`图标顶部显示一个数字。 这个源选项卡是你定期去更新 GitHub 上的文件的地方。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:40]:
```suggestion
在上一节中，你对页面内容进行了更改。 这意味着你的项目文件已更改，VS Code 应在“源”菜单 `Source Control` 图标顶部显示一个数字。 这个源选项卡是你定期去更新 GitHub 上的文件的地方。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+
+1. 单击 VS Code 中的“源代码管理”选项卡`Source Control`以查看已更改文件的列表。 如果你看到需要安装 `git` 的消息，请按照提供的说明进行操作，然后重新加载 VS Code。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:43]:
```suggestion
1. 单击 VS Code 中的“源代码管理”选项卡 `Source Control` 以查看已更改文件的列表。如果你看到需要安装 `git` 的消息，请按照提供的说明进行操作，然后重新加载 VS Code。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+2. 单击提交消息上方的<kbd>•••</kbd>“3个点”的菜单并选择<kbd>Remote</kbd> > <kbd>Add Remote</kbd>。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:45]:
```suggestion
2. 单击提交消息上方的 <kbd>•••</kbd> “3个点”的菜单并选择 <kbd>Remote</kbd> > <kbd>Add Remote</kbd>。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+3. 选择 <kbd>Add remote from GitHub</kbd>。 如有必要，请执行任何身份验证步骤，然后返回 VS Code 并重复此操作。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:47]:
```suggestion
3. 选择 <kbd>Add remote from GitHub</kbd>。如有必要，请执行任何身份验证步骤，然后返回 VS Code 并重复此操作。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+4. 你应该在 GitHub 上看到所有仓库的列表。 选择你为此项目创建的那个。 如果你没有看到你的项目，请直接粘贴其 GitHub URL。 你可能还会被要求为该仓库指定一个本地名称。 可以选择任何你喜欢的名称。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:49]:
```suggestion
4. 你应该在 GitHub 上看到所有仓库的列表。选择你为此项目创建的那个。如果你没有看到你的项目，请直接粘贴其 GitHub URL。你可能还会被要求为该仓库指定一个本地名称。可以选择任何你喜欢的名称。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+5. 在菜单窗格的顶部，将有一个地方可以输入**提交消息**`commit message`（你的文件更改的描述）。 输入 `initial commit` 并按 <kbd>Commit</kbd> 按钮提交这些更改。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:51]:
```suggestion
5. 在菜单窗格的顶部，将有一个地方可以输入**提交消息** `commit message`（你的文件更改的描述）。输入 `initial commit` 并按 <kbd>Commit</kbd> 按钮提交这些更改。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+7. 你可能会看到一条消息，告诉你没有`staged`提交，并询问你是否要暂存它们。 单击<kbd>Always</kbd> 并继续。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:53]:
```suggestion
6. 你可能会看到一条消息，告诉你没有”staged“提交，并询问你是否要暂存它们。单击 <kbd>Always</kbd> 并继续。
```

Wrong serial number, the same below.

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+8. 最后，更改文件列表应替换为<kbd>Publish</kbd> 按钮。 单击此按钮将你提交的更改发送到 GitHub。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:55]:
```suggestion
7. 最后，更改文件列表应替换为 <kbd>Publish</kbd> 按钮。 单击此按钮将你提交的更改发送到 GitHub。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+
+### 在 GitHub 上查看你的项目
+
+要验证你的项目是否已成功存储在 GitHub 上，请访问 GitHub.com 并在你的帐户下查看你的仓库列表。 选择你创建的新文件，并确认它包含你的 Astro 项目文件。
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:60]:
```suggestion
要验证你的项目是否已成功存储在 GitHub 上，请访问 GitHub.com 并在你的帐户下查看你的仓库列表。选择你创建的新文件，并确认它包含你的 Astro 项目文件。
```

File: src/content/docs/zh-cn/tutorial/1-setup/4.mdx [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+## 任务清单
+
+<Checklist>
+- [ ] 我把我的项目储存到了 GitHub 上。
+</Checklist>
+</Box>
+
+### 资源
+
+- <p>[在 VS Code 中使用 Git 源代码管理`Source control`](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>external</Badge></p>
```
Comment (luoingly) [inline src/content/docs/zh-cn/tutorial/1-setup/4.mdx:73]:
```suggestion
- <p>[在 VS Code 中使用 Git 源代码管理](https://code.visualstudio.com/docs/sourcecontrol/overview#_git-support) <Badge>external</Badge></p>
```

Unnecessary (I think).

**Review by luoingly — CHANGES_REQUESTED** [review]
Hi @KurumiFrost ~ Thanks for your contributions! 💜

Your translation looks very good to me, but there are many problems in the details. Some problems are important, such as wrong project number serial number, some problems may just be my personal obsession (punctuation, space, style etc.) 🤣. I only reviewed part of the content, if I have time I will continue.

Of course, I also want to know if you will continue to translate this tutorial?

**Comment by KurumiFrost** [comment]
> Hi @KurumiFrost ~ Thanks for your contributions! 💜
> 
> Your translation looks very good to me, but there are many problems in the details. Some problems are important, such as wrong project number serial number, some problems may just be my personal obsession (punctuation, space, style etc.) 🤣. I only reviewed part of the content, if I have time I will continue.
> 
> Of course, I also want to know if you will continue to translate this tutorial?

Hi~ @luoingly! Thank you for your feedback and kind words~ I apologize for any problems or inaccuracies in the translation. I understand that some details, such as project numbers and serial numbers, may have been incorrect. I appreciate your attention to detail and your obsession with punctuation, spacing, and style.

Regarding the question about continuing the translation of the tutorial, I'll do my best to assist you. However, I must let you know that I'm currently prioritizing exam preparation, so my availability for contributing might be limited. If you have any urgent translation requests, please let me know, and I'll try to accommodate them as much as possible💙

**Comment by luoingly** [comment]
> Hi~ @luoingly! Thank you for your feedback and kind words~ I apologize for any problems or inaccuracies in the translation. I understand that some details, such as project numbers and serial numbers, may have been incorrect. I appreciate your attention to detail and your obsession with punctuation, spacing, and style.
> 
> Regarding the question about continuing the translation of the tutorial, I'll do my best to assist you. However, I must let you know that I'm currently prioritizing exam preparation, so my availability for contributing might be limited. If you have any urgent translation requests, please let me know, and I'll try to accommodate them as much as possible💙

Thank you for your understanding! ✨

I completely understand that you have limited time recently. I also understand that exam preparation may be taking up most of your time. Good luck to you! ~

If could, I'll help you with your contribution when I have some free time. Please keep an eye on the PR requests for your repo. Thank you for your contribution! 💜

**Comment by luoingly** [comment]
I've made my review suggestions into commits here: https://github.com/luoingly/astro-docs/tree/patch-2

If you like, you could sync your fork first, and then I will open a PR from my branch into your repo. 🚀

**Comment by KurumiFrost** [comment]
> I've made my review suggestions into commits here: https://github.com/luoingly/astro-docs/tree/patch-2
> 
> If you like, you could sync your fork first, and then I will open a PR from my branch into your repo. 🚀

Sure, I have already synced my fork. Please go ahead. 🚀

**Pull request closed without merging**